src/core/logical_layers/utils/applyNewLayerSource.ts

```diff
@@ -0,0 +1,20 @@
+import { store } from '~core/store/store';
+import { enabledLayersAtom } from '../atoms/enabledLayers';
+import { createUpdateLayerActions } from './createUpdateActions';
+import type { LayerSource } from '../types/source';
+import type { Action } from '@reatom/core-v2';
+
+export function applyNewLayerSource(newSource: LayerSource) {
```

Comment:
Shouldn't this function be an action inside one of the atoms?

Reply:
I could use some help refactoring here.

Originally it was part of `applyNewMCDAConfig()` as an action inside `mcdaLayer` atom. But accessing mcdaLayer from `MCDALayerEditor` caused a circular dependency, that's why I put it into util function instead. (also, does the layer update itself look correct?)

@dqunbp @vkozio

Reply:
ok, keep it here